godot-codegen/src/godot_version.rs

```diff
@@ -28,25 +29,44 @@ pub struct GodotVersion {
 
 pub fn parse_godot_version(version_str: &str) -> Result<GodotVersion, Box<dyn Error>> {
     let regex = Regex::new(
-        //  major  minor     [patch]                                              official|custom_build|gentoo
-        //  v      v         v                                                    v
-        r#"(\d+)\.(\d+)(?:\.(\d+))?\.(alpha|beta|dev|rc|stable)[0-9]*\.(?:mono\.)?(?:[a-z_]+\.([a-f0-9]+)|official)"#,
+        r#"(?x) # ignore whitespace and allow line comments (starting with `#`)
+        ^
+        (?P<major>\d+)
+        \.(?P<minor>\d+)
+        (?:\.(?P<patch>\d+))?
+        \.(?P<stability>alpha|beta|dev|rc|stable)[0-9]*
+        (\.[^.]+)+? # mono|official|custom_build|gentoo|arch_linux|...
```

Comment:
What's the meaning of `...` here? Is it not 3 arbitrary characters?

Reply:
See [how the version string is built up](https://github.com/godot-rust/gdext/issues/118#issuecomment-1465748123). Turns out my regex captures the format quite well, but this was entirely by accident :)

A version string like `4.0.stable.foo.bar.baz-1.2.3.mono.mybuild.abcde1234` is actually valid, and can happen if people create their own custom build or link in third-party modules.

Reply:
Thanks a lot for analyzing this! We should probably document this format somewhere... just where?

I also saw now that you used `+?`, the lazy operator -- so the Git revision afterwards should always be parsed correctly, if available?

Maybe it makes sense to add some exotic versions to the tests 🙂

Reply:
Based on my analysis, I even made the `status` (formerly `stability`) field accept `[^.]+`. Though we parse it explicitly, we currently don't use it anyway.

Added a comment linking to https://github.com/godot-rust/gdext/issues/118#issuecomment-1465748123 so it's documented at least somewhere.